conda/cli/conda_argparse.py

```diff
@@ -104,44 +107,51 @@ def generate_parser():
     configure_parser_init(sub_parsers)
     configure_parser_install(sub_parsers)
     configure_parser_list(sub_parsers)
+    configure_parser_notices(sub_parsers)
     configure_parser_package(sub_parsers)
     configure_parser_remove(sub_parsers, aliases=["uninstall"])
     configure_parser_rename(sub_parsers)
     configure_parser_run(sub_parsers)
     configure_parser_search(sub_parsers)
     configure_parser_update(sub_parsers, aliases=["upgrade"])
-    configure_parser_notices(sub_parsers)
+    configure_parser_plugins(sub_parsers, p.plugin_subcommands)
 
     return p
 
 
-def do_call(arguments: argparse.Namespace, parser: ArgumentParser):
+def do_call(args: argparse.Namespace, parser: ArgumentParser):
     """
     Serves as the primary entry point for commands referred to in this file and for
     all registered plugin subcommands.
     """
+    # disable all external plugins if requested
     if context.no_plugins:
         context.plugin_manager.disable_external_plugins()
 
-    # First, check if this is a plugin subcommand; if this attribute is present then it is
-    if getattr(arguments, "plugin_subcommand", None):
-        context.plugin_manager.invoke_pre_commands(arguments.plugin_subcommand.name)
-        result = arguments.plugin_subcommand.action(sys.argv[2:])
-        context.plugin_manager.invoke_post_commands(arguments.plugin_subcommand.name)
-
-        return result
-
-    relative_mod, func_name = arguments.func.rsplit(".", 1)
-    # func_name should always be 'execute'
-    from importlib import import_module
-
-    module = import_module(relative_mod, __name__.rsplit(".", 1)[0])
-
-    command = relative_mod.replace(".main_", "")
-    context.plugin_manager.invoke_pre_commands(command)
-    result = getattr(module, func_name)(arguments, parser)
-    context.plugin_manager.invoke_post_commands(command)
-
+    # let's see if during the parsing phase it was discovered that the
+    # called command was in fact a plugin subcommand
+    plugin_subcommand = getattr(args, "plugin_subcommand", None)
+
+    if plugin_subcommand:
+        # pass on the rest of the plugin specific args or fall back to
+        # the whole discovered arguments
+        try:
+            plugin_args = tuple(args.plugin_args)
+        except AttributeError:
+            plugin_args = args
+        context.plugin_manager.apply_pre_commands(plugin_subcommand.name)
+        result = plugin_subcommand.action(plugin_args)
+        context.plugin_manager.apply_post_commands(plugin_subcommand.name)
```

Comment:
```suggestion
        context.plugin_manager.invoke_pre_commands(plugin_subcommand.name)
        result = plugin_subcommand.action(plugin_args)
        context.plugin_manager.invoke_post_commands(plugin_subcommand.name)
```

Reply:
Okay, I'm backing away from the computer now...